src/Groups/recog.jl

```diff
@@ -0,0 +1,118 @@
+@attributes mutable struct GroupRecognitionNode{T <: GAPGroup}
+  input_group::T
+  gap_node::GapObj
+
+  function GroupRecognitionNode{T}(G::GAPGroup, gap_node::GapObj) where T
+     res = new{T}(G, gap_node)
+     return res
+  end
+end
+
+GapObj(node::GroupRecognitionNode) = node.gap_node
+
+# did the recognition not fail?
+is_ready(node::GroupRecognitionNode) = GAP.Globals.IsReady(node.gap_node)
+
+# does the node describe a leaf?
+is_leaf(node::GroupRecognitionNode) = GAP.Globals.IsLeaf(node.gap_node)
+
+# the group from which `node` was constructed
+input_group(node::GroupRecognitionNode) = node.input_group
+
+# the group stored in `node`, provided that recognition did not fail;
+# the `GapObj`s of its elements store memory information
+function group(node::GroupRecognitionNode)
+  @req is_ready(node) "recognition failed, no group stored"
+  return _oscar_group(GAP.Globals.Grp(node.gap_node))
+end
+
+function Base.show(io::IO, node::GroupRecognitionNode)
+  if is_terse(io)
+    print(io, "Recognition node of a group")
```

Comment:
Maybe "Recognition tree" would be more apt and less confusing for users? (I've contemplate changing this name in `recog` itself but since there it'd require changing a lot of code, I've put it off for now)